pkg/operator/controllers/machine/machine_controller_test.go

```diff
@@ -176,7 +176,7 @@ func TestMachineValid(t *testing.T) {
 			errs := r.machineValid(ctx, tt.machine, isMaster)
```

Comment:
Because it is a proper `Reconciler` now we should test it via `Reconcile` method. Otherwise we leave out some code without coverage.

Reply:
Agreed, I'm working on improving the unit test and will follow up.

Reply:
@m1kola @jewzaam @mjudeikis the original `machinechecker_test` was only meant to check machineValid, and to account for all test cases here would add quite a bit more to this PR. Additionally, many operator controllers don't have unit tests at all, and so I'm wondering if the unit test improvements proposed here should instead go into a new user story for improving/adding unit tests across all operator controllers.

Reply:
I think what @m1kola is saying that we should be Testing `Reconcile` instead of `machineValid` . So for now lets just rewrite same tests for to be testing using Reconcile (See monitoring controller for example). And if you could raise story to improve testing in general as follow-up

Reply:
I think we should improve this test while we are refactoring this codebase. We can deal with others later (as MJ suggested - please raise a story for this).

> to account for all test cases here would add quite a bit more to this PR.

Yes, it will be a bit of extra work, but you will not have to cover extra cases here. You will need to change how you instantiate `MachineReconciler` in the test, add some mocks (for `r.arocli` at least) and replace test assertion `if !reflect.DeepEqual(errs, tt.wantErrs)` which a check that looks into actual conditions on the cluster object.

Let me know if you need help with this.

Reply:
@bennerv has assisted me in rewriting the unit test to test Reconcile rather than machineValid (same test cases as what was there previously). Let me know if this looks good.